# Neptune 2016

## Introduction

The course will include morning workshops on general computing skills based on the book [Practical Computing for Biologists](http://www.sinauer.com/practical-computing-for-biologists.html). These will be taught be [Casey Dunn](http://dunnlab.org), along with August Guang (a grad student in Casey's lab) with help from additional TAs. Morning workshops will be followed by afternoon workshops on specific applications of bioinformatics.

A few general notes about the course structure:

- Many details of the course are specific to UNIX-like operating systems, which include Linux and Apple OS X.

- Computationally intensive analyses will be conducted remotely on [Amazon Web Services](https://aws.amazon.com) cloud computers.

- We will also use UNIX for local command line, programing, and data processing exercises. This means that it will not be possible to use Microsoft Windows alone for most course activities. If your primary laptop is Microsoft Windows, there are a couple options. We suggest that you borrow a Mac laptop for the course. Alternatively, if you prefer you can install Linux on a Virtual Machine and run it from within Windows.

- Course content, schedule, code, and other files will be distributed on [github](https://github.com). Course materials are available at [neptune2016](https://github.com/neptune2016).

- Course communication and coordination will take place on [slack](https://slack.com). Casey will send a link for you to join our slack team.


## To do before you arrive

- Fill out the [course survey](http://goo.gl/forms/nfuwpCX74x). This will be used to tailor the content to your interests and make decisions about how to run the course to best fit everyones' needs. *Please complete the survey by May 6, 2016*.

- Bring a copy of [Practical Computing for Biologists](http://www.sinauer.com/practical-computing-for-biologists.html) (abbreviated in this document as PCfB). Complete the reading for the first class (see Schedule below) prior to arrival. Also download an familiarize yourself with the [example files](http://practicalcomputing.org) provided at the [book website](http://practicalcomputing.org).

- Bring an Ethernet adapter and Ethernet cable, as wifi access will be slow.

- Create a [github](https://github.com) account. If you use an academic email address, you may be eligible for free upgrades.

- Create an [Amazon](http://www.amazon.com) account. You won't need to purchase anything for this class, but you will need a valid account to get access to the Amazon cloud computers we will use.

- After you fill out the course survey, Casey will send you an invitation to the slack team.

- Install the required software listed below. 

- Instructors, please see [instructors.md](https://github.com/neptune2016/syllabus/blob/master/instructors.md) for additional preparation items.


## Required software

Please install the following tools on your laptop before you arrive. This will reduce network load during the course, and also allow us to get started right away.

### All operating systems

- [Sublime](https://www.sublimetext.com/3) text editor version 3

- [cyberduck](https://cyberduck.io/?l=en) file transfer tool (or equivalent if you have Linux)

### OS X

- OS X command line developer tools. To see if you have them installed, open /Applications/Utilities/Terminal and type `gcc` at the command line. If `gcc` runs, you are probably all set. If it asks you if you would like to install the Developer tools, follow the instructions to do so.

### Windows

- [PuTTY](http://www.putty.org) ssh client

- Install [Virtual Box](https://www.virtualbox.org/wiki/Downloads), and then create an Ubuntu virtual machine. See Practical Computing for Biologists page 458 for instructions on how to do this, as well as the shared folder instructions at the [book website](http://practicalcomputing.org/ubuntu). This will allow you to run Linux from within windows.

## Instructions

### Using Amazon Web Services

### Git

## Schedule

For the complete schedule, please see the [program](.program.pdf?raw=true). This syllabus focuses on the 6 classes to be taught by Casey. These classes are from 9.00 - 13.00 on each of the following days.

The lectures are a bit out of sequence with the book to be sure you have the skills needed for the other course activities led by other instructors.

### Class 1 (Tuesday May 31): Getting Set Up, The Unix Command Line
> Reading: Haddock & Dunn "Before You Begin", Chapters 1, 4-5, Appendix 1, 3 

### Application Lecture: Sequencing project planning (Tuesday May 31 14.00-15.30)
> Reading: [Dunn et al. 2013](http://dx.doi.org/10.1093/icb/ict068). Suggested optional reading: [Guang et al. 2016](http://dx.doi.org/10.1016/j.tree.2015.12.007)

### Application Practical (Tuesday May 31 16.00-18.00) - Regular Exressions
> Reading: Haddock & Dunn Chapters 2-3, Appendix 2

### Class 2 (Thursday June 2): Command Line (continued), Remote Machines, Version Control
> Reading: Haddock & Dunn Chapters 6, 20, Haddock & Dunn new git chapter (to be distributed at the course)


### Class 3 (Friday June 3): Programming
> Reading: Haddock & Dunn Chapters 7-9

### Class 4 (Monday June 6): Programming continued
> Reading: Haddock & Dunn Chapters 10-12

### Application Practical: building a phylogenetic tree (Monday June 6, 16.00-18.00)
Run by Max and Casey

### Class 5 (Tuesday June 7): Graphics, Visualizing Data, Project Management, High Performance Computing
> Reading: Haddock & Dunn Chapters 17-19

### Class 6 (Wednesday June 8): To Be Determined

## Reading

> Haddock and Dunn 2011. [Practical Computing for Biologists](http://www.sinauer.com/practical-computing-for-biologists.html). Sinauer and Associates.

## Other resources
